### PR TITLE
PLUGIN-279 default join on null keys to false

### PR DESCRIPTION
**Pull request by yaojiefeng**

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java [hunk 1]
```diff
@@ -254,7 +254,7 @@ boolean requiredPropertiesContainMacros() {
   }
 
   boolean isNullSafe() {
-    return joinNullKeys == null ? true : joinNullKeys;
+    return joinNullKeys == null ? false : joinNullKeys;
```
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java:257]: This change is backwards incompatible right?

Comment (yaojiefeng) [reply core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java:257]: Yes, this is intended, look at https://cdap.atlassian.net/browse/PLUGIN-279 for more details

Comment (CuriousVini) [reply core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java:257]: As a best practice, we always make backwards incompatible changes in major versions. 
Why do we need exception in this case?

Comment (sreevatsanraman) [reply core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java:257]: +1

**Comment by yaojiefeng** [comment]
After discussion, decided to move this out of 6.3.0, so closing this one

File: core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java (hunk 1 above)
Comment (nitinmotgi) [reply core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java:257]: Backward incompatible is for all major features. Not for all tiny things that affect users. If this issue was priortized then we need to figure out how to make it without affecting the users. It's not a blanket rule for only making backward incompatible changes in major version. They are mostly for wire-protocol, deprecation and such. @CuriousVini @yaojiefeng @sreevatsanraman

**Pull request closed without merging**